src/autocomplete/databases/yql/helpers.ts

```diff
@@ -20,6 +21,36 @@ export const tokenDictionary: TokenDictionary = {
 type AnyRuleInList = (rules: number | number[]) => boolean;
 type AllRulesInList = (rules: number[]) => boolean;
 
+function isFirstPreviousTokenOfType(
+    tokenStream: TokenStream,
+    dictionary: TokenDictionary,
+    tokenIndex: number,
+    tokenType: number,
+): boolean | undefined {
+    const firstIndex = tokenIndex - 1;
+
+    let token = tokenStream.get(firstIndex);
+
+    if (token.type === tokenType) {
+        return true;
+    }
+
+    let currentIndex = firstIndex - 1;
+
+    while (token?.type === dictionary.SPACE) {
+        const found = tokenStream.get(currentIndex);
+
+        if (found.type === tokenType) {
+            return true;
+        }
+        token = found;
```

Comment:
why do we need `token` variable, if it's always the same as `found`? also I'd rename `found` to `foundToken` or better yet `currentToken`

Reply:
I refactored it with `do ... while` loop, so it is only `token` remained